src/test/groovy/pl/allegro/tech/build/axion/release/infrastructure/git/GitRepositoryTest.groovy

```diff
@@ -146,6 +148,31 @@ class GitRepositoryTest extends Specification {
         then:
         tags.tags == ['release-1']
     }
+    
+    def "should return all tagged commits matching the pattern provided"() {
+        given:
+        repository.tag('release-1')
+        repository.commit(['*'], "commit after release-1")
+        repository.tag('release-2')
+        repository.commit(['*'], "commit after release-2")
+        repository.tag('another-tag-1')
+        repository.commit(['*'], "commit after another-tag-1")
+        repository.commit(['*'], "commit after another-tag-1-2")
+        repository.tag('release-4')
+        repository.commit(['*'], "commit after release-4")
+        repository.tag('release-3')
+        repository.commit(['*'], "commit after release-3")
+    
+        when:
+        List<TagsOnCommit> allTaggedCommits  = repository.allTaggedCommits(~/^release.*/, null, true, false)
+    
+        then:
+        allTaggedCommits.size() == 4
+        allTaggedCommits.get(0).tags.get(0).equals('release-3')
```

Comment:
oh, yes, but you can transform this by `allTaggedCommits.collect( { t -> tags[0] } ) == ['release-3', ...]` (i might be bit off cos i didn't check the actual code, but you should get the idea)